### PR TITLE
fix async callback table

### PR DESCRIPTION
**Pull request by cdetrio**
Passing a few more VM tests with this fix.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -503,11 +507,8 @@ exports.buildModule = function (funcs, imports = [], exports = [], cbs = []) {
   (memory 500)
   (export "memory" (memory 0))
 
-  (table
-    (export "callback")
-    anyfunc
-    (elem ${cbs.join(' ')})
-  )
-   ${funcStr}
+  ${callbackTableStr}
```
Comment (axic) [inline index.js]: This changes the callbacks from being a table into exported functions. These are two different things. (Not saying the old version worked properly.)

Comment (cdetrio) [reply index.js]: Ah, thanks. ewasm-kernel didn't expect a table, I should've fixed it there originally.

File: index.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ exports.evm2wast = function (evmCode, opts = {
   // to figure out what .wast files to include
   const opcodesUsed = new Set()
   const ignoredOps = new Set(['JUMP', 'JUMPI', 'JUMPDEST', 'POP', 'STOP', 'INVALID'])
-  const callbackTable = []
+  let callbackTable = []
```
Comment (axic) [inline index.js]: Oh wow, lint didn't complain that `.push` is called on a `const`?:)

File: index.js [hunk 3]
```diff
@@ -477,14 +477,19 @@ exports.resolveFunctions = function (funcSet) {
...
+exports.buildModule = function (funcs, imports = [], callbacks = []) {
   let funcStr = ''
   for (let func of funcs) {
     funcStr += func
   }
-  for (let exprt of exports) {
-    funcStr += `(export "${exprt}" (func $${exprt}))`
-  }
+
+  let callbackTableStr = `
```
Comment (axic) [inline index.js]: Wonder if this works if `callback` has no elements. Do you want to make it conditional on that?

**Comment by axic** [comment]
Needs rebase.

**Comment by cdetrio** [comment]
Down to 76 failures with https://github.com/ewasm/ewasm-kernel/pull/7

**Pull request merged**